tests/unit/LSYiiValidatorsTest.php

```diff
@@ -27,4 +27,155 @@ public function testHtmlEntityDecodeFilter()
             $this->assertEquals($expected, $actual);
         }
     }
+
+    /**
+     * Testing that the xssfilter attribute will always be
+     * false for super admin users even if filterxsshtml is
+     * changed.
+     */
+    public function testXssFilterAttributeForSuperAdmin()
+    {
+        //Mocking super admin login.
+        \Yii::app()->session['loginID'] = 1;
+        $superAdminValidator = new \LSYii_Validators();
+
+        $this->assertTrue(\Yii::app()->getConfig('filterxsshtml'), 'filterxsshtml should be true by default.');
+        $this->assertFalse($superAdminValidator->xssfilter, 'The xssfilter attribute should be false for super admins.');
+
+        //Changing filterxsshtml.
+        \Yii::app()->setConfig('filterxsshtml', false);
+        $newSuperAdminValidator = new \LSYii_Validators();
+
+        $this->assertFalse(\Yii::app()->getConfig('filterxsshtml'), 'filterxsshtml was just changed to false.');
+        $this->assertFalse($newSuperAdminValidator->xssfilter, 'The xssfilter attribute should be false for super admins.');
+
+        //Returning to original values.
+        \Yii::app()->setConfig('filterxsshtml', true);
+        \Yii::app()->session['loginID'] = null;
+    }
+
+    /**
+     * Testing that the xssfilter attribute varies
+     * for regular users depending on filterxsshtml.
+     */
+    public function testXssFilterAttributeForRegularUsers()
+    {
+        //Create user.
+        $newPassword = createPassword();
+        $userId = \User::insertUser('test_user', $newPassword, 'John Doe', 1, 'jd@mail.com');
```

Comment:
I like a random namle for user just in case 

(i already have test, test_rc etc … in my dev instance) 

see https://github.com/LimeSurvey/LimeSurvey/blob/a3605854a63d56b360660dea47e144ea766d382f/tests/functional/backend/XssAndScriptCheckTest.php#L22